compiler/src/dmd/targetcompiler.d

```diff
@@ -0,0 +1,211 @@
+/**
+ * Encapsulates the vagaries of which of the three D compilers (gdc, ldc or dmd) is being built
+ *
+ * Copyright:   Copyright (C) 1999-2025 by The D Language Foundation, All Rights Reserved
+ * Authors:     $(LINK2 https://www.digitalmars.com, Walter Bright)
+ * License:     $(LINK2 https://www.boost.org/LICENSE_1_0.txt, Boost License 1.0)
+ * Source:      $(LINK2 https://github.com/dlang/dmd/blob/master/compiler/src/dmd/mars.d, _targetcompiler.d)
+ * Documentation:  https://dlang.org/phobos/dmd_targetcompiler.html
+ * Coverage:    https://codecov.io/gh/dlang/dmd/src/master/compiler/src/dmd/targetcompiler.d
+ */
+
+module dmd.targetcompiler;
+
+version (IN_GCC) {}        // compiler is being built with gdc
+else version (IN_LLVM) {}  // compiler is being built with ldc
+else version = MARS;       // default means compiler is built with Digital Mars compiler (DMD)
+
+/***************************
+ */
+
+version (IN_GCC)
+{
+    enum TargetCompiler = "GNU D";    // compiler being built
+    enum SwitchPrefix = "f";          // prefix to switch
+    enum SwitchVariadic = "fno-rtti"; // disables D-style variadic functions
+    enum const(char)* SwitchExceptions = "fno-exceptions"; // switch that disables exceptions
+    enum const(char)* SwitchScopeGuard = "fno-exceptions"; // switch that disables exceptions
+}
+else version (IN_LLVM)
+{
+    enum TargetCompiler = "LDC";
+    enum SwitchPrefix = "";
+    enum SwitchVariadic = "betterC";
```

Comment:
```suggestion
    enum SwitchVariadic = "fno-rtti";
```
A total misnomer IMO btw, this is for when the user doesn't want any TypeInfos, which is much more than just for variadics.